app/printing/printFull/get.controller.js

```diff
@@ -1,13 +1,75 @@
 const { logger } = require('../../../common/logging/logger')
-const { removeUrlLevels } = require('../../../common/utils/util')
+const { removeUrlLevels, groupBy, getYearMonthFromDate, getStatusText } = require('../../../common/utils/util')
+const { getCommentText } = require('../../../common/utils/getCommentText')
+const { getSentencePlan } = require('../../../common/data/sentencePlanningApi')
+const { COMMENT_TYPES, ACTION_STATUS_TYPES } = require('../../../common/utils/constants')
 
-const printFullSentencePlan = async ({ path, params: { id, planId } }, res) => {
+const printFullSentencePlan = async ({ path, params: { id, planId }, tokens }, res) => {
   try {
     const backUrl = removeUrlLevels(path, 1)
-    res.render(`${__dirname}/index`, { backUrl })
+
+    let objectives
+    let comments
+
+    // get the sentence plan
+    try {
+      ;({ comments, objectives = [] } = await getSentencePlan(planId, tokens))
+    } catch (error) {
+      logger.error(`Could not retrieve active sentence plan ${planId} for offender ${id}, error: ${error}`)
+      return res.render('app/error', { error })
+    }
+
+    // determine the status of each objective
+    objectives = objectives.map(objective => {
```

Comment:
This should be a _forEach_ and not a _map_. Map returns a new Array, but we are actually mutating the original objective objects.